eth/AccountManager.cpp

```diff
@@ -0,0 +1,230 @@
+/*
+	This file is part of cpp-ethereum.
+
+	cpp-ethereum is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	cpp-ethereum is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/** @file AccountManager.cpp
+ * @author Yann <yann@ethdev.com>
+ * @date 2016
+ */
+
+#include <libdevcore/SHA3.h>
+#include <libdevcore/FileSystem.h>
+#include <libethcore/ICAP.h>
+#include <libethcore/KeyManager.h>
+#include "AccountManager.h"
+using namespace std;
+using namespace dev;
+using namespace dev::eth;
+
+bool AccountManager::execute(int argc, char** argv)
+{
+	if ((string)argv[1] == "wallet")
+	{
+		if (4 < argc && (string)argv[2] == "import")
+		{
+			keyManager();
+			string file =  argv[3];
+			string name = argv[4];
```

Comment:
if the keyManager is instanciated but the wallet never opened (using the Master password), everything will be read and imported as bare keys.
So we could never open the wallet if we want to be closer than how geth manage keys.